sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareDirectoryAsyncClient.java

```diff
@@ -134,6 +134,9 @@ public ShareServiceVersion getServiceVersion() {
      */
     public ShareFileAsyncClient getFileClient(String fileName) {
         String filePath = directoryPath + "/" + fileName;
+        if (directoryPath.isEmpty()) {
+            filePath = fileName;
```

Comment:
Can you add a comment that this is to accommodate the root?

Reply:
done